### PR TITLE
Fixes for #5038

### PR DESCRIPTION
**Pull request by mjmadsen**
We were using the deltas to simulate human behavior and shouldn't be applied to altitude.

### CODE REVIEW

**Comment by oralunal** [comment]
Polyline Walker has real problems mjm

**Comment by mjmadsen** [comment]
Yeah, I saw @Anakin5 s comments about the polywalker. Any chance you can take a gander at that one?

**Comment by mjmadsen** [comment]
Set poly to fetch the last known alt if it cannot get it through self.polyline.get_alt

Still need to add the pause/unpause to prevent it from teleporting.

**Comment by Anakin5** [comment]
Can you rollback the changes on Polyline while you come up with a fix ? For now it is not usable at all and some of us use it in our config.
Thank you

**Comment by mjmadsen** [comment]
@Anakin5 Could you try this, I'm at work and catch test anything :-1:

**Comment by solderzzc** [comment]
@mjmadsen  The polyline walker is not working as expect, right side is dev branch, left side is master branch.
screen shot 2016-09-03 at 12 06 28 am

**Comment by mjmadsen** [comment]
Ugh.

**Comment by mjmadsen** [comment]
@solderzzc Was that with the recent polyline_walker I had up? I had to revert because of some conflicts.

File: pokemongo_bot/walkers/polyline_walker.py [hunk 1]
```diff
@@ -15,6 +15,6 @@ def __init__(self, bot, dest_lat, dest_lng):
                                                               (self.dest_lat, self.dest_lng),
                                                               self.speed, google_map_api_key=self.bot.config.gmapkey)
         self.pol_lat, self.pol_lon = self.polyline.get_pos()
-        self.pol_alt = self.polyline.get_alt() or uniform(self.bot.config.alt_min, self.bot.config.alt_max)
+        self.pol_alt = self.polyline.get_alt() or self.alt
```
Comment (th3w4y) [inline pokemongo_bot/walkers/polyline_walker.py]: There is already PR https://github.com/PokemonGoF/PokemonGo-Bot/pull/5135 for this

File: pokemongo_bot/walkers/polyline_walker.py [hunk 2]
```diff
@@ -15,6 +15,6 @@ def __init__(self, bot, dest_lat, dest_lng):
...
         super(PolylineWalker, self).__init__(self.bot, self.pol_lat, self.pol_lon,
                                              self.pol_alt, fixed_speed=True)
```
Comment (th3w4y) [inline pokemongo_bot/walkers/polyline_walker.py]: here we need fixed_speed=self.speed

File: pokemongo_bot/walkers/polyline_walker.py (hunk 1 above)
Comment (mjmadsen) [inline pokemongo_bot/walkers/polyline_walker.py]: #5132 < #5135 

I can pull this entire PR down if you'd prefer to go off of yours

**Comment by th3w4y** [comment]
there are some other extra fixes that you were working on here... I was referring only at the .get_alt() part...

File: pokemongo_bot/walkers/step_walker.py [hunk 3]
```diff
@@ -27,7 +27,7 @@ def __init__(self, bot, dest_lat, dest_lng, dest_alt=None, fixed_speed=False):
             
         if fixed_speed:
             # PolylineWalker uses a fixed speed!
-            self.speed = self.bot.config.walk_min
+            self.speed = fixed_speed
```
Comment (th3w4y) [inline pokemongo_bot/walkers/step_walker.py]: This needs a bit of change in the __init__ also then...

Comment (th3w4y) [inline pokemongo_bot/walkers/step_walker.py]: I took this over in https://github.com/PokemonGoF/PokemonGo-Bot/pull/5135

**Comment by th3w4y** [comment]
@mjmadsen 

This is not needed...
>Still need to add the pause/unpause to prevent it from teleporting.

**Comment by mjmadsen** [comment]
As long as we can get it working right :+1:

**Comment by th3w4y** [comment]
@mjmadsen remaining taks not taken over somewhere else from this PR are:


- the  alt_delta
- the random_lat_long (factor 5..)

@mjmadsen will you do them in a different PR?

**Comment by mjmadsen** [comment]
Can do. What did you put in for the alt delta range? If it's small I could see it being somewhat more human like (swinging arms when walking). #5140

**Comment by th3w4y** [comment]
@mjmadsen that is purely in StepWalker... i did not change..

**Comment by mjmadsen** [comment]
@th3w4y Took a look at it and it's a variance of .4m. I don't think that's unreasonable so I won't bother removing it from StepWalker

**Comment by th3w4y** [comment]
@mjmadsen I agree

**Comment by th3w4y** [comment]
@solderzzc Thanks for your screenshot...

I believe the bug was in the cache validation.. https://github.com/PokemonGoF/PokemonGo-Bot/pull/5135/commits/8f7f22c90e92ccafafa4956ba006624c3f90ab5e 

I have fix in the PR https://github.com/PokemonGoF/PokemonGo-Bot/pull/5135

**Pull request closed without merging**